test/project_types/extension/features/argo/base_test.rb

```diff
@@ -0,0 +1,253 @@
+            assert_equal(config[:renderer_version], "0.4.3")
+          end
+        end
+
+        def test_config_aborts_with_error_if_run_list_command_fails
+          Base64.stubs(:strict_encode64).returns(@encoded_script)
+          with_stubbed_script(@context, Argo::Base::SCRIPT_PATH) do
+            ShopifyCli::JsSystem.any_instance.stubs(:package_manager).returns('yarn')
+            Argo::Base.any_instance.stubs(:run_yarn_install).returns(true)
+            Argo::Base.any_instance.stubs(:run_yarn_run_script).returns(true)
+            ShopifyCli::JsSystem.any_instance.stubs(:call).returns([@result, @error_message, mock(success?: false)])
+            error = assert_raises(ShopifyCli::Abort) { @dummy_argo.config(@context) }
+            assert error
+              .message
+              .include?(
+                @context.message(
+                  'features.argo.dependencies.argo_missing_renderer_package_error',
+                  @error_message
+                )
+              )
+          end
+        end
+
+        def test_config_aborts_with_error_when_no_version_found
+          Base64.stubs(:strict_encode64).returns(@encoded_script)
+          with_stubbed_script(@context, Argo::Base::SCRIPT_PATH) do
+            ShopifyCli::JsSystem.any_instance.stubs(:package_manager).returns('yarn')
+            Argo::Base.any_instance.stubs(:run_yarn_install).returns(true)
+            Argo::Base.any_instance.stubs(:run_yarn_run_script).returns(true)
+            ShopifyCli::JsSystem.any_instance.stubs(:call).returns([@result, @no_error, mock(success?: true)])
+            Argo::Base.any_instance.stubs(:find_version_tag).returns(nil)
+
+            error_message = 'The renderer package version is not a valid SemVer Version (http://semver.org)'
+            error = assert_raises(ShopifyCli::Abort) { @dummy_argo.config(@context) }
+            assert error
+              .message
+              .include?(
+                @context.message(
+                  'features.argo.dependencies.argo_renderer_package_invalid_version_error',
+                  error_message
+                )
+              )
+          end
+        end
+
+        def test_config_aborts_with_error_if_found_version_is_invalid
+          Base64.stubs(:strict_encode64).returns(@encoded_script)
+          invalid_version = "2.invalid.0"
+          with_stubbed_script(@context, Argo::Base::SCRIPT_PATH) do
+            ShopifyCli::JsSystem.any_instance.stubs(:package_manager).returns('yarn')
+            Argo::Base.any_instance.stubs(:run_yarn_install).returns(true)
+            Argo::Base.any_instance.stubs(:run_yarn_run_script).returns(true)
+            ShopifyCli::JsSystem.any_instance.stubs(:call).returns([@result, @no_error, mock(success?: true)])
+            Argo::Base.any_instance.stubs(:find_version_tag).returns(invalid_version)
+
+            error_message = 'The renderer package version is not a valid SemVer Version (http://semver.org)'
```

Comment:
Same here

Reply:
In this case, `with_stubbed_script` method creates a script file. Do you think it is still good to include the following: File.stubs(:exist?).returns(true)?